x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/search_knn_query_vector_builder.yml

```diff
@@ -8,6 +8,9 @@ setup:
       indices.create:
         index: index-with-embedded-text
         body:
+          settings:
+            index:
+              number_of_shards: 1
```

Comment:
I think instead of  `number_of_shards`, instead of `num_candidates:3` on line 121, we should ask for `size:3`.  This  guarantees to produce 3 final hits.

Reply:
this was the first thing I tried, but it doesn't seem to work.

Reply:
@tteofili this is because the test is measuring `total_hits`. This is the number of documents considered a "match" but not the number of documents returned. 

```
  - match: { hits.total.value: 3 }
```

You should instead verify the doc_id of the closest vectors. Total hits of the knn query are effected by the shard count as we will match `num_shard*num_candidate` vectors.

Reply:
thanks @benwtrent , I'll directly check for exact `doc_ids`

Reply:
good point, for just the size of returned total hits we can use:   - length: { hits.hits: 3 }